pyxem/signals/diffraction1d.py

```diff
@@ -118,43 +112,7 @@ def get_virtual_image(self, left, right):
 
         return vdfim
 
-    def as_lazy(self, *args, **kwargs):
-        """Create a copy of the Diffraction1D object as a
-        :py:class:`~pyxem.signals.diffraction1d.LazyDiffraction1D`.
-
-        Parameters
-        ----------
-        copy_variance : bool
-            If True variance from the original Diffraction1D object is copied to
-            the new LazyDiffraction1D object.
-
-        Returns
-        -------
-        res : :py:class:`~pyxem.signals.diffraction1d.LazyDiffraction1D`.
-            The lazy signal.
-        """
-        res = super().as_lazy(*args, **kwargs)
-        res.__class__ = LazyDiffraction1D
-        res.__init__(**res._to_dictionary())
-        return res
-
-    def decomposition(self, *args, **kwargs):
-        super().decomposition(*args, **kwargs)
-        self.__class__ = Diffraction1D
-
 
 class LazyDiffraction1D(LazySignal, Diffraction1D):
 
```

Comment:
Please could you just explain why this will work? Are we inheriting all of this stuff from Signal1D now?

Reply:
Yes, this is exactly the idea and the `signal_type` will be set correctly when a new signal is  instantiated inside a function for example, so there is no need anymore to change the `signal_type`. This behaviour is covered by the test.